packages/cojson/src/coValues/coMap.ts

```diff
@@ -197,17 +229,13 @@ export class RawCoMapView<
    * @category 1. Reading */
   keys<K extends keyof Shape & string = keyof Shape & string>(): K[] {
     return (Object.keys(this.latest) as K[]).filter((key) => {
-      const latest = this.latest[key];
-      if (!latest) {
-        return undefined;
-      }
+      const latestChange = this.latest[key];
 
-      const includeUntil = this.atTimeFilter;
-      const lastEntry = includeUntil
-        ? this.getOps()[key]?.findLast((entry) => entry.madeAt <= includeUntil)
-        : latest!;
+      if (!latestChange) {
```

Comment:
Optimized to use `latest` since it's already processed considering the "atTime" filter